### ISSUE TITLE
tests hang

### ISSUE THREAD

**Issue by ZeroChaos-** [CoreSecurity/impacket#59]
In the gentoo ebuild we run tests by invoking "nosetests" in the impacket directory.  Here is the output:

>>> Test phase: dev-python/impacket-9999
 * python2_7: running distutils-r1_run_phase python_test
test_message_construction (impacket.testcases.ImpactPacket.test_ICMP6.TestICMP6) ... ok
test_message_decoding (impacket.testcases.ImpactPacket.test_ICMP6.TestICMP6) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.002s

OK
test_creation (impacket.testcases.ImpactPacket.test_IP6.TestIP6)
Test IP6 Packet creation. ... ok
test_decoding (impacket.testcases.ImpactPacket.test_IP6.TestIP6)
Test IP6 Packet decoding. ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.000s

OK
test_compressions (impacket.testcases.ImpactPacket.test_IP6_Address.TestIP6_Address)
Test IP6 Address compressions. ... ok
test_construction (impacket.testcases.ImpactPacket.test_IP6_Address.TestIP6_Address)
Test IP6 Address construction ... ok
test_conversions (impacket.testcases.ImpactPacket.test_IP6_Address.TestIP6_Address)
Test IP6 Address conversions. ... ok
test_scoped_addresses (impacket.testcases.ImpactPacket.test_IP6_Address.TestIP6_Address)
Test scoped addresses. ... ok
test_unicode_representation (impacket.testcases.ImpactPacket.test_IP6_Address.TestIP6_Address)
Test IP6 Unicode text representations ... ok

----------------------------------------------------------------------
Ran 5 tests in 0.001s

OK
test_add_option_to_dest_opts (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_add_option_to_hop_by_hop (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_chained_basic_options (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_chained_basic_options_inside_ipv6_packet (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_create_simple_dest_opts (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_create_simple_hop_by_hop (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_create_simple_routing_options (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_decoding_chained_basic_options_inside_ipv6_packet (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_decoding_extension_header_from_string (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_decoding_multi_option_destination_options (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_decoding_multi_option_hop_by_hop (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_decoding_simple_destination_options (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_decoding_simple_hop_by_hop (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_decoding_simple_routing_options (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_pad_dest_opts_when_adding_option (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_pad_hop_by_hop_when_adding_option (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_simple_dest_opts_contained_in_ipv6 (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_simple_hop_by_hop_contained_in_ipv6 (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok
test_simple_routing_options_contained_in_ipv6 (impacket.testcases.ImpactPacket.test_IP6_Extension_Headers.TestIP6) ... ok

----------------------------------------------------------------------
Ran 19 tests in 0.003s

OK
test_01 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP get_packet ... ok
test_02 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP getters ... ok
test_03 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP port setters ... ok
test_04 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP offset setters ... ok
test_05 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP win setters ... ok
test_06 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP checksum setters ... ok
test_07 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP flags setters ... ok
test_08 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP reset_flags ... ok
test_09 (impacket.testcases.ImpactPacket.test_TCP.TestTCP)
Test TCP set_flags ... ok

----------------------------------------------------------------------
Ran 9 tests in 0.001s

OK
.....................................EEEEEEEEEEEEE

At that point it seems to hang forever doing nothing.

**Comment by asolino** [comment]
Hey there.. 

I'm not familiar with `nosetests`. If you provide a way to reproduce this problem you're having I might be able to help.

**Comment by ZeroChaos-** [comment]
What is the proper way to run tests at build time?  Perhaps I'm simply doing it wrong.

**Comment by asolino** [comment]
The current tests are not that automated sadly.

You can run [runalltestcases.sh](https://github.com/CoreSecurity/impacket/blob/master/impacket/testcases/ImpactPacket/runalltestcases.sh) and [runalltestcases.sh](https://github.com/CoreSecurity/impacket/blob/master/impacket/testcases/dot11/runalltestcases.sh)
for the ImpactPacket and dot11 features.

For the SMB_RPC part the history is different. 

* SMB 1/2/3 related tests: you need to configure [test_smb.py](https://github.com/CoreSecurity/impacket/blob/master/impacket/testcases/SMB_RPC/test_smb.py), the data for every dialect tested. Also have in mind the target machine for SMB3 must be different than for the rest of the dialects tested (see the notes at the beginning of the file).
The hashes and aesKeys can be grabbed if you run [secretsdump.py].(https://github.com/CoreSecurity/impacket/blob/master/examples/secretsdump.py)
* DCERPC relates tests: you need to fill out the configuration [dcetests.cfg](https://github.com/CoreSecurity/impacket/blob/master/impacket/testcases/SMB_RPC/dcetests.cfg) and the run [rundce.sh](https://github.com/CoreSecurity/impacket/blob/master/impacket/testcases/SMB_RPC/rundce.sh). 

Keep in mind the target machine must be a Windows 7 SP1, joined to a domain controller (so the Kerberos tests pass). With other target machines, some RPC calls might fail.

So, in a nutshell that's what I do for testing the library before building. A lot of improvements to do of course, help is welcome! :)

Hope this helps

**Comment by ZeroChaos-** [comment]
thanks, that helps a lot.  It looks like both the ImpactPacket and dot11 tests work when run within the gentoo package manager, so I'll use those to ensure the software is working.  Thanks for the help.
